test/framework/src/main/java/org/elasticsearch/test/AbstractWireSerializingTestCase.java

```diff
@@ -61,10 +61,6 @@ public void testEqualsAndHashcode() throws IOException {
 
             T secondInstance = copyInstance(firstInstance);
             assertEquals("instance is not equal to self", secondInstance, secondInstance);
-            if (false == firstInstance.equals(secondInstance)) {
-                firstInstance.equals(secondInstance);
-            assertEquals("instance is not equal to its copy", firstInstance, secondInstance);
-            }
```

Comment:
Good catch, I wanted only to remove what looked like a debug statement, I pushed a fix.